modules/dag-shared/src/main/scala/org/tessellation/dag/types/DAGBlock.scala

```diff
@@ -0,0 +1,16 @@
+package org.tessellation.dag.types
+
+import cats.data.NonEmptyList
+
+import org.tessellation.schema.height.Height
+import org.tessellation.schema.transaction.Transaction
+import org.tessellation.security.signature.Signed
+
+import derevo.circe.magnolia.{decoder, encoder}
+import derevo.derive
+import io.estatico.newtype.ops._
+
+@derive(encoder, decoder)
+case class DAGBlock(transactions: Set[Signed[Transaction]], parent: NonEmptyList[BlockReference]) {
+  val height: Height = Height(parent.sortBy(_.height.value).head.height.coerce + 1L)
```

Comment:
`parent.maxBy(_.height.value)`

Reply:
If there is a `minBy` I will use it!

Reply:
There is :)

Also `coerce` is safer because you are not coupled with the name of constant encapsulated by newtype